hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/ha/BootstrapStandby.java

```diff
@@ -405,8 +423,14 @@ private boolean checkLogsAvailableForRead(FSImage image, long imageTxId,
     }
   }
 
-  private boolean checkLayoutVersion(NamespaceInfo nsInfo) throws IOException {
-    return (nsInfo.getLayoutVersion() == HdfsServerConstants.NAMENODE_LAYOUT_VERSION);
+  private boolean checkLayoutVersion(NamespaceInfo nsInfo, boolean isRollingUpgrade) {
+    if (isRollingUpgrade) {
+      // During a rolling upgrade the service layout versions may be different,
+      // but we should check that the layout version being sent is compatible
+      return nsInfo.getLayoutVersion() <=
+          HdfsServerConstants.MINIMUM_COMPATIBLE_NAMENODE_LAYOUT_VERSION;
```

Comment:
Shouldn't the comparison here be `>=`, to validate that the layoutVersion is at least the minimum compatible one.

Reply:
The version numbers are negative so the `HdfsServerConstants.MINIMUM_COMPATIBLE_NAMENODE_LAYOUT_VERSION` is -61. We want the nsInfo's layout version to be a "higher" version than the minimum which is -61. So we need to use `<=` because a "higher" version is a lower negative like -67.

Reply:
Got you. Thanks for clarifying.